# Coffee Sales Analysis

![coffee_image](./assets/coffee-image.jpeg "alt-coffee_image")

## About Author

Author: Joshua Farara

Project: title

### Contact Info
Click on link below to contact/follow/correct me:

Email: joshua.farara@gmail.com

[LinkedIn](https://www.linkedin.com/in/joshuafarara/)

[Facebook](https://www.facebook.com/josh.farara/)

[Twitter](https://x.com/FararaTheArtist)

[Github](https://github.com/JoshuaFarara)


## Import Libraries

We will use the following libraries¶
1. Pandas: Data manipulation and analysis
2. Numpy: Numerical operations and calculations
3. Matplotlib: Data visualization and plotting
4. Seaborn: Enhanced data visualization and statistical graphics
5. Scipy: Scientific computing and advanced mathematical operations

In [15]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import matplotlib.pyplot as plt
import seaborn as sns
import scipy as sp

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

## Data Loading and Exploration | Cleaning

### Load a CSV file then creating a dataframe

In [16]:
# Kaggle Notebook
# df = pd.read_csv('/kaggle/input/coffee-sales/index.csv')


#Local Machine Notebook
df = pd.read_csv('data/coffee_sales_data.csv')

### Set the option to show maximum columns

In [17]:
pd.set_option('display.max_columns', None) 
pd.set_option('display.max_rows', None)

### Get a sneak peek of data
The purpose of a sneak peek is to get a quick overview of the data and identify any potential problems or areas of interest

In [18]:
df.head(5)

date                 datetime cash_type                 card  money  \
0  2024-03-01  2024-03-01 10:15:50.520      card  ANON-0000-0000-0001   38.7   
1  2024-03-01  2024-03-01 12:19:22.539      card  ANON-0000-0000-0002   38.7   
2  2024-03-01  2024-03-01 12:20:18.089      card  ANON-0000-0000-0002   38.7   
3  2024-03-01  2024-03-01 13:46:33.006      card  ANON-0000-0000-0003   28.9   
4  2024-03-01  2024-03-01 13:48:14.626      card  ANON-0000-0000-0004   38.7   

     coffee_name  
0          Latte  
1  Hot Chocolate  
2  Hot Chocolate  
3      Americano  
4          Latte

### Let's see the column names

In [19]:
df.columns

Index(['date', 'datetime', 'cash_type', 'card', 'money', 'coffee_name'], dtype='object')

### Let's have a look on the shape of the dataset

In [20]:
print(f"The Number of Rows are {df.shape[0]}, and columns are {df.shape[1]}.")

The Number of Rows are 896, and columns are 6.


### Let's have a look on the columns and their data types using detailed info function

In [21]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 896 entries, 0 to 895
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   date         896 non-null    object 
 1   datetime     896 non-null    object 
 2   cash_type    896 non-null    object 
 3   card         807 non-null    object 
 4   money        896 non-null    float64
 5   coffee_name  896 non-null    object 
dtypes: float64(1), object(5)
memory usage: 42.1+ KB


### Count the missing values

In [22]:
df.isnull().sum()

date            0
datetime        0
cash_type       0
card           89
money           0
coffee_name     0
dtype: int64

### Observation Set 1

* There are 896 rows, and 6 columns in the dataset.

* The data type of all columns are objects except for df['money'] which is float.

* The columns in the datasets are:
    * 'date', 'datetime', 'cash_type', 'card', 'money', 'coffee_name'
    
* There are a few missing values in the dataset, which we will read in detail and deal with later on in the notebook.

#### Changing column names 

Changed: cash_type to payment_type since cash and card payments are accepted

In [23]:
df.rename(columns={'cash_type':'payment_type'}, inplace=True)

In [24]:
df.columns

Index(['date', 'datetime', 'payment_type', 'card', 'money', 'coffee_name'], dtype='object')

## Changing Data Types of Columns

In [25]:
df

date                 datetime payment_type                 card  \
0    2024-03-01  2024-03-01 10:15:50.520         card  ANON-0000-0000-0001   
1    2024-03-01  2024-03-01 12:19:22.539         card  ANON-0000-0000-0002   
2    2024-03-01  2024-03-01 12:20:18.089         card  ANON-0000-0000-0002   
3    2024-03-01  2024-03-01 13:46:33.006         card  ANON-0000-0000-0003   
4    2024-03-01  2024-03-01 13:48:14.626         card  ANON-0000-0000-0004   
5    2024-03-01  2024-03-01 15:39:47.726         card  ANON-0000-0000-0005   
6    2024-03-01  2024-03-01 16:19:02.756         card  ANON-0000-0000-0006   
7    2024-03-01  2024-03-01 18:39:03.580         card  ANON-0000-0000-0007   
8    2024-03-01  2024-03-01 19:22:01.762         card  ANON-0000-0000-0008   
9    2024-03-01  2024-03-01 19:23:15.887         card  ANON-0000-0000-0008   
10   2024-03-01  2024-03-01 19:29:17.391         card  ANON-0000-0000-0009   
11   2024-03-02  2024-03-02 10:22:06.957         card  ANON-0000-0000-0010   
12   2024-03-02  2024-03-02 10:30:35.668         cash                  NaN   
13   2024-03-02  2024-03-02 10:41:41.249         card  ANON-0000-0000-0011   
14   2024-03-02  2024-03-02 11:59:45.484         card  ANON-0000-0000-0012   
15   2024-03-02  2024-03-02 14:38:35.535         card  ANON-0000-0000-0013   
16   2024-03-02  2024-03-02 16:37:24.475         card  ANON-0000-0000-0014   
17   2024-03-02  2024-03-02 17:34:54.969         card  ANON-0000-0000-0004   
18   2024-03-03  2024-03-03 10:10:43.981         cash                  NaN   
19   2024-03-03  2024-03-03 10:27:18.561         card  ANON-0000-0000-0015   
20   2024-03-03  2024-03-03 11:33:56.118         card  ANON-0000-0000-0016   
21   2024-03-03  2024-03-03 12:26:56.098         card  ANON-0000-0000-0012   
22   2024-03-03  2024-03-03 13:09:36.321         card  ANON-0000-0000-0017   
23   2024-03-03  2024-03-03 17:06:40.271         card  ANON-0000-0000-0018   
24   2024-03-03  2024-03-03 17:08:45.895         card  ANON-0000-0000-0019   
25   2024-03-03  2024-03-03 18:03:23.369         card  ANON-0000-0000-0020   
26   2024-03-03  2024-03-03 18:04:27.946         card  ANON-0000-0000-0020   
27   2024-03-03  2024-03-03 18:08:04.959         card  ANON-0000-0000-0020   
28   2024-03-04  2024-03-04 10:03:51.994         card  ANON-0000-0000-0001   
29   2024-03-04  2024-03-04 10:54:50.958         card  ANON-0000-0000-0021   
30   2024-03-04  2024-03-04 11:05:16.184         card  ANON-0000-0000-0012   
31   2024-03-04  2024-03-04 14:04:37.734         card  ANON-0000-0000-0022   
32   2024-03-05  2024-03-05 09:59:52.651         card  ANON-0000-0000-0001   
33   2024-03-05  2024-03-05 14:34:55.963         card  ANON-0000-0000-0023   
34   2024-03-05  2024-03-05 17:34:06.043         card  ANON-0000-0000-0024   
35   2024-03-05  2024-03-05 17:35:25.021         card  ANON-0000-0000-0025   
36   2024-03-05  2024-03-05 17:36:28.571         card  ANON-0000-0000-0026   
37   2024-03-05  2024-03-05 17:37:13.659         card  ANON-0000-0000-0027   
38   2024-03-05  2024-03-05 17:38:09.354         card  ANON-0000-0000-0009   
39   2024-03-05  2024-03-05 17:56:15.776         card  ANON-0000-0000-0028   
40   2024-03-05  2024-03-05 18:01:31.242         card  ANON-0000-0000-0028   
41   2024-03-06  2024-03-06 12:30:27.089         cash                  NaN   
42   2024-03-06  2024-03-06 13:24:07.667         card  ANON-0000-0000-0012   
43   2024-03-06  2024-03-06 13:25:14.351         card  ANON-0000-0000-0012   
44   2024-03-06  2024-03-06 14:52:01.761         card  ANON-0000-0000-0029   
45   2024-03-06  2024-03-06 14:53:18.344         card  ANON-0000-0000-0029   
46   2024-03-07  2024-03-07 10:08:58.945         cash                  NaN   
47   2024-03-07  2024-03-07 10:18:40.543         card  ANON-0000-0000-0030   
48   2024-03-07  2024-03-07 11:03:58.976         card  ANON-0000-0000-0031   
49   2024-03-07  2024-03-07 11:25:43.977         cash                  NaN   
50   2024-03-07  2024-03-07 15:40